### PR TITLE
Followups for method call error change

### PR DESCRIPTION
**Pull request by jackh726**
Each commit is self-contained. Fixes most of the followup reviews from that PR.

r? @estebank

### CODE REVIEW

File: compiler/rustc_typeck/src/check/callee.rs [hunk 1]
```diff
@@ -171,7 +171,14 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
...
                         },
                     );
-                    return Some(CallStep::DeferredClosure(closure_sig));
+                    if let Ok(_) = std::env::var("DEBUG") {
+                        dbg!(def_id);
+                        dbg!(self.tcx.hir().get_if_local(def_id));
+                        if let Some(def_span) = self.tcx.def_ident_span(def_id) {
+                            dbg!(def_span);
+                        }
+                    }
```
Comment (estebank) [inline compiler/rustc_typeck/src/check/callee.rs:180]: Can we remove this?

File: compiler/rustc_typeck/src/check/fn_ctxt/checks.rs [hunk 2]
```diff
@@ -1782,3 +1714,48 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
...
+        match tcx.hir().get_if_local(def_id) {
+            Some(hir::Node::Expr(hir::Expr {
+                kind: hir::ExprKind::Closure(_, decl, _, span, ..),
+                ..
+            })) => {
+                let mut spans: MultiSpan = (*span).into();
+
+                for input in decl.inputs {
+                    spans.push_span_label(input.span, String::new());
+                }
```
Comment (estebank) [inline compiler/rustc_typeck/src/check/fn_ctxt/checks.rs:1762]: It seems to me the way that the closure header and the arg spans overlap looks bad. Removing this for now should be "good enough".

**Review by estebank — APPROVED** [review]
Thank you for the clearly separated commits and apologies for the delay. r=me after addressing nitpicks and rebasing.

File: compiler/rustc_typeck/src/check/callee.rs (hunk 1 above)
Comment (jackh726) [reply compiler/rustc_typeck/src/check/callee.rs:180]: 🤦‍♂️how did I leave this

**Comment by estebank** [comment]
It seems you have to rebase again 😅

**Comment by jackh726** [comment]
Yeah. I had to step away from my computer before I was actually able to rebase (just made the review changes).

**Pull request merged**